api/sso_test.go

```diff
@@ -0,0 +1,520 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	jwt "github.com/golang-jwt/jwt"
+	"github.com/netlify/gotrue/conf"
+	"github.com/netlify/gotrue/models"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type SSOTestSuite struct {
+	suite.Suite
+	API      *API
+	Config   *conf.GlobalConfiguration
+	AdminJWT string
+}
+
+func TestSSO(t *testing.T) {
+	api, config, err := setupAPIForTest()
+	require.NoError(t, err)
+
+	ts := &SSOTestSuite{
+		API:    api,
+		Config: config,
+	}
+	defer api.db.Close()
+
+	if config.SAML.Enabled {
+		suite.Run(t, ts)
+	}
+}
+
+func (ts *SSOTestSuite) SetupTest() {
+	models.TruncateAll(ts.API.db)
+
+	claims := &GoTrueClaims{
+		Role: "supabase_admin",
+	}
+	token, err := jwt.NewWithClaims(jwt.SigningMethodHS256, claims).SignedString([]byte(ts.Config.JWT.Secret))
+	require.NoError(ts.T(), err, "Error generating admin jwt")
+
+	ts.AdminJWT = token
+}
+
+func (ts *SSOTestSuite) TestNonAdminJWT() {
+	// TODO
```

Comment:
Not sure if this was forgotten - should we remove? or keep as placeholder for later

Reply:
Yeah keep as a placeholder for later. Some more extensive end-to-end tests are needed and that's about 1000+ more LOC so I'd rather keep this short.